src/main/java/ch/njol/skript/expressions/ExprBannerItem.java

```diff
@@ -0,0 +1,145 @@
+import org.bukkit.block.banner.PatternType;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.*;
+
+@Name("Banner Pattern Item")
+@Description({
+	"Gets the item from a banner pattern type.",
+	"NOTE: Not all banner pattern types have an item.",
+})
+@Examples({
+	"set {_item} to creeper charged banner pattern item",
+	"set {_item} to snout banner pattern item",
+	"set {_item} to thing banner pattern item"
+})
+@Since("INSERT VERSION")
+public class ExprBannerItem extends SimpleExpression<ItemType> {
+
+	private static Map<Object, Material> bannerMaterials = new HashMap<>();
+	private static boolean PATTERN_TYPE_IS_REGISTRY = false;
+
+	static {
+		Registry<PatternType> patternRegistry = Bukkit.getRegistry(PatternType.class);
+		Object[] bannerPatterns;
+		if (patternRegistry != null) {
+			bannerPatterns = patternRegistry.stream().toArray();
+			PATTERN_TYPE_IS_REGISTRY = true;
+		} else {
+			try {
+				Class<?> patternClass = Class.forName("org.bukkit.block.banner.PatternType");
+				if (patternClass.isEnum()) {
+					//noinspection unchecked,rawtypes
+					Class<? extends Enum> enumClass = (Class<? extends Enum>) patternClass;
+					bannerPatterns = enumClass.getEnumConstants();
+				} else {
+					throw new IllegalStateException("PatternType is neither an enum nor a valid registry.");
+				}
+			} catch (ClassNotFoundException e) {
+				throw new RuntimeException(e);
+			}
+		}
+		if (bannerPatterns != null) {
+			for (Object object : bannerPatterns) {
+				Material material = getMaterial(object);
+				if (material != null)
+					bannerMaterials.put(object, material);
+			}
+			Skript.registerExpression(ExprBannerItem.class, ItemType.class, ExpressionType.COMBINED,
+				"[a[n]] %*bannerpatterntypes% item[s]");
+		}
+	}
+
+	private PatternType[] patternTypes;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		//noinspection unchecked
+		patternTypes = ((Literal<PatternType>) exprs[0]).getArray();
+		for (PatternType type : patternTypes) {
+			if (!bannerMaterials.containsKey(type)) {
+				Skript.error("There is no item for the banner pattern type '" + type + "'.");
+				return false;
+			}
+		}
+		return true;
+	}
+
+	@Override
+	protected ItemType @Nullable [] get(Event event) {
+		List<ItemType> itemTypes = new ArrayList<>();
+		for (PatternType type : patternTypes) {
+			Material material = bannerMaterials.get(type);
+			ItemType itemType = new ItemType(material);
+			itemTypes.add(itemType);
+		}
+		return itemTypes.toArray(new ItemType[0]);
+	}
+
+	@Override
+	public boolean isSingle() {
+		return patternTypes.length == 1;
+	}
+
+	@Override
+	public Class<ItemType> getReturnType() {
+		return ItemType.class;
+	}
+
+	@Override
+	public String toString(@Nullable Event event, boolean debug) {
+		return StringUtils.join(patternTypes, ",") + " items";
```

Comment:
![image](https://github.com/user-attachments/assets/810e87a8-6ca7-4686-a071-525093a151ca)
this doesn't work either.
You should probably just be storing the literal and calling tostring on it, or manually calling Classes.toString